src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -66,39 +78,81 @@
+  const exhangeFollower = (newTopicLists, willFollow, topicId) => {
+    const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
+    newTopicLists[indexTopic].following = !!willFollow;
+    newTopicLists[indexTopic].user_id_follower = myId;
+    return newTopicLists[indexTopic];
+  };
+
+  const handleTopic = (from, willFollow, item, index) => {
     if (from === FROM_FOLLOWED_TOPIC_INITIAL) {
       const newFollowedTopics = [...topics];
-      newFollowedTopics[index].following = !!willFollow;
-      newFollowedTopics[index].user_id_follower = myId;
-      DiscoveryAction.setDiscoveryInitialTopics(newFollowedTopics, discoveryDispatch);
+      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
+
+      DiscoveryAction.setDiscoveryInitialTopics(
+        discovery.initialTopics.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        }),
+        discoveryDispatch
+      );
     }
 
     if (from === FROM_UNFOLLOWED_TOPIC_INITIAL) {
       const newFollowedTopics = [...topics];
-      newFollowedTopics[index].following = !!willFollow;
-      newFollowedTopics[index].user_id_follower = myId;
+      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
 
-      DiscoveryAction.setDiscoveryInitialTopics(newFollowedTopics, discoveryDispatch);
+      DiscoveryAction.setDiscoveryInitialTopics(
+        discovery.initialTopics.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        }),
+        discoveryDispatch
+      );
     }
 
     if (from === FROM_FOLLOWED_TOPIC) {
-      const newFollowedTopics = [...followedTopic];
-      newFollowedTopics[index].following = !!willFollow;
-      newFollowedTopics[index].user_id_follower = myId;
+      const newFollowedTopics = [...newMapFollowedTopics];
+      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
 
-      setFollowedTopic(newFollowedTopics);
+      setFollowedTopic(
+        followedTopic.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        })
+      );
     }
 
     if (from === FROM_UNFOLLOWED_TOPIC) {
-      const newUnFollowedTopic = [...unfollowedTopic];
-      newUnFollowedTopic[index].following = !!willFollow;
-      newUnFollowedTopic[index].user_id_follower = myId;
+      const newUnFollowedTopic = [...newMapUnfollowedTopics];
+      const newTopic = exhangeFollower(newUnFollowedTopic, willFollow, item.topic_id);
 
-      setUnfollowedTopic(newUnFollowedTopic);
+      setUnfollowedTopic(
+        unfollowedTopic.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        })
+      );
     }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleTopic` function has a lot of repeated code for different cases. This is another violation of the DRY principle. Consider creating a helper function to handle the common logic.
```diff
-  const handleTopic = (from, willFollow, item, index) => {
-    if (from === FROM_FOLLOWED_TOPIC_INITIAL) {
-      const newFollowedTopics = [...topics];
-      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
-      DiscoveryAction.setDiscoveryInitialTopics(
-        discovery.initialTopics.map((topic) => {
-          if (topic.topic_id === newTopic.topic_id) {
-            return newTopic;
-          }
-          return topic;
-        }),
-        discoveryDispatch
-      );
-    }
-    // ... similar code for other cases ...
-  };

+  const updateTopics = (topics, newTopic) => {
+    return topics.map((topic) => {
+      if (topic.topic_id === newTopic.topic_id) {
+        return newTopic;
+      }
+      return topic;
+    });
+  };
+
+  const handleTopic = (from, willFollow, item, index) => {
+    let newTopics;
+    if (from === FROM_FOLLOWED_TOPIC_INITIAL || from === FROM_UNFOLLOWED_TOPIC_INITIAL) {
+      newTopics = [...topics];
+    } else if (from === FROM_FOLLOWED_TOPIC) {
+      newTopics = [...newMapFollowedTopics];
+    } else if (from === FROM_UNFOLLOWED_TOPIC) {
+      newTopics = [...newMapUnfollowedTopics];
+    }
+    const newTopic = exhangeFollower(newTopics, willFollow, item.topic_id);
+    const updatedTopics = updateTopics(topics, newTopic);
+    DiscoveryAction.setDiscoveryInitialTopics(updatedTopics, discoveryDispatch);
+  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->